widget/embedded/src/components/WalletDerivationPathModal/WalletDerivationPathModal.tsx

```diff
@@ -0,0 +1,157 @@
+import type { PropTypes } from './WalletDerivationPathModal.types';
+import type { DerivationPath } from '@rango-dev/wallets-shared';
+
+import { i18n } from '@lingui/core';
+import {
+  Button,
+  Divider,
+  Image,
+  MessageBox,
+  Select,
+  TextField,
+} from '@rango-dev/ui';
+import React, { useEffect, useMemo, useState } from 'react';
+
+import { WIDGET_UI_ID } from '../../constants';
+import { namespaces } from '../../constants/namespaces';
+import { WatermarkedModal } from '../common/WatermarkedModal';
+import {
+  LogoContainer,
+  Spinner,
+  WalletImageContainer,
+} from '../WalletModal/WalletModalContent.styles';
+
+import {
+  derivationPathInputStyles,
+  InputLabel,
+  InputsContainer,
+} from './WalletDerivationPathModal.styles';
+
+const customDerivationPath: DerivationPath = {
+  id: 'custom',
+  label: 'Custom',
+  generateDerivationPath: (index: string) => index,
+};
+
+export function WalletDerivationPathModal(props: PropTypes) {
+  const { onClose, onConfirm, selectedNamespace, image, type } = props;
+
+  const [selectedDerivationPath, setSelectedDerivationPath] =
+    useState<DerivationPath | null>(null);
+  const [derivationPathIndex, setDerivationPathIndex] = useState('0');
+
+  const {
+    isOpen,
+    derivationPaths,
+  }: { derivationPaths: DerivationPath[]; isOpen: boolean } = useMemo(() => {
```

Comment:
I'm not sure why you are memoizing this part of code. let's keep it simple and don't add unwanted closure behaviors that generally happens.

Reply:
Tnx, Fixed.